admin/class-smaily-for-wp-admin.php

```diff
@@ -73,15 +73,21 @@ public function enqueue_scripts() {
 	 * @since 3.0.0
 	 */
 	public function smaily_admin_render() {
-		global $wpdb;
-
 		// Create admin template.
 		$template = new Smaily_For_WP_Template( 'admin/partials/smaily-for-wp-admin-page.php' );
 
 		// Load configuration data.
-		$table_name = esc_sql( $wpdb->prefix . 'smaily_config' );
-		$data       = $wpdb->get_row( "SELECT * FROM `$table_name` LIMIT 1" );
-		$template->assign( (array) $data );
+		$api_credentials = get_option( 'smailyforwp_api_option' );
+		$form_options    = get_option( 'smailyforwp_form_option' );
+
+		$credentials_set = ! empty( $api_credentials['subdomain'] ) && ! empty( $api_credentials['username'] ) && ! empty( $api_credentials['password'] );
```

Comment:
With fresh install `smailyforwp_api_option` and `smailyforwp_form_option` are not set and `get_option` will return `NULL`.

I suggest you switch to getter pattern to always have normalized configuration option values:
```php
private function get_api_credentials() {
    $credentials = get_option( 'smailyforwp_api_option', array() );
    return array_merge( array(
        'subdomain' => null,
        'username' => null,
        'password' => null,
    ), $credentials );
}
```

This way you don't need to do `isset` checks before trying to access credentials key, as it will be always set, you'd just need to ensure the value is what you'd expect it to be.

Reply:
`get_option` should return `false`

Reply:
Added an option handler, which implements this getter pattern. 2b03272